src/UITests/AIWinSession.cs

```diff
@@ -60,20 +60,49 @@ private void LaunchApplicationAndAttach()
             var configPathArgument = $"--ConfigFolder \"{Path.Combine(TestContext.TestResultsDirectory, TestContext.TestName)}\"";
 
             process = Process.Start(exePath, configPathArgument);
-            process.WaitForInputIdle();
 
-            // small buffer between splash screen disappearing 
-            // and main window initializing; otherwise in rare
-            // cases splash screen can be picked up as main window
-            Thread.Sleep(30000);
-
-            DesiredCapabilities appCapabilities = new DesiredCapabilities();
-            appCapabilities.SetCapability("appTopLevelWindow", process.MainWindowHandle.ToString("x"));
-            session = new WindowsDriver<WindowsElement>(new Uri(WindowsApplicationDriverUrl), appCapabilities);
+            const int attempts = 10; // this number should give us enough retries for the build to work
+            RetryAttemptToStartNewSession(attempts);
```

Comment:
I wonder if it would make sense to just call it `StartNewSession` with retrying being abstracted into that.  🔕

Reply:
Renamed

Reply:
Renamed